### PR TITLE
[WIP] [doc] Metric learning tutorial

### PR DESCRIPTION
**Pull request by tuofeilunhifi**
1. add cub metric learning tutorial
2. modify cls_config
3. modify modelzoo_cls link

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -14,11 +14,11 @@ repos:
     hooks:
       - id: isort
         exclude: thirdparty/|examples/|pai_jobs/
-  - repo: https://github.com/pre-commit/mirrors-yapf
-    rev: v0.30.0
-    hooks:
-      - id: yapf
-        exclude: thirdparty/|examples/
+  # - repo: https://github.com/pre-commit/mirrors-yapf
```
Comment (wenmengzhou) [inline .pre-commit-config.yaml:17]: do not comment this

File: configs/classification/resnet/cifar_resnet50.py [hunk 2]
```diff
@@ -1,15 +1,9 @@
-_base_ = '../../base.py'
+_base_ = './resnet50_b32x8_100e_jpg.py'
```
Comment (wenmengzhou) [inline configs/classification/resnet/cifar_resnet50.py:1]: 1. cifar10 training from scratch may not need 8 gpus
2. if use only download this configfile, can we use it to train successfully

File: configs/metric_learning/imagenet_resnet50_1000kid_jpg.py [hunk 3]
```diff
@@ -31,19 +31,19 @@
...
-data_test_list = 'data/imagenet_raw/meta/val_labeled.txt'
-data_test_root = 'data/imagenet_raw/'
-data_all_list = 'data/imagenet_raw/meta/all_labeled.txt'
-data_root = 'data/imagenet_raw/'
+data_train_list = '/apsarapangu/disk1/yunji.cjy/imagenet_raw/meta/train_labeled.txt'
+data_train_root = '/apsarapangu/disk1/yunji.cjy/imagenet_raw/train'
+data_test_list = '/apsarapangu/disk1/yunji.cjy/imagenet_raw/meta/val_labeled.txt'
+data_test_root = '/apsarapangu/disk1/yunji.cjy/imagenet_raw/val'
+data_all_list = '/apsarapangu/disk1/yunji.cjy/imagenet_raw/meta/all_labeled.txt'
+data_root = '/apsarapangu/disk1/yunji.cjy/imagenet_raw/'
```
Comment (wenmengzhou) [inline configs/metric_learning/imagenet_resnet50_1000kid_jpg.py:46]: remove path prefix

File: configs/metric_learning/imagenet_resnet50_1000kid_jpg.py [hunk 4]
```diff
@@ -97,7 +97,7 @@
 
 # learning policy
 lr_config = dict(policy='step', step=[30, 60, 90])
-checkpoint_config = dict(interval=10)
+checkpoint_config = dict(interval=1)
```
Comment (wenmengzhou) [inline configs/metric_learning/imagenet_resnet50_1000kid_jpg.py:100]: is it too frequent to save checkpoint?

File: docs/source/tutorials/metric_learning.md [hunk 5]
```diff
@@ -0,0 +1,195 @@
+# metric learning tutorial
```
Comment (wenmengzhou) [inline docs/source/tutorials/metric_learning.md:1]: also provide a notebook  similar to the ones under tutorial

File: docs/source/tutorials/metric_learning.md [hunk 6]
```diff
@@ -0,0 +1,195 @@
...
+
+</details>
+
+**Examples:**
+
+Edit `data_root`path in the `${CONFIG_PATH}` to your own data path.
+
+    single gpu training:
+    ```shell
+    CUDA_VISIBLE_DEVICES=0 python -m torch.distributed.launch --nproc_per_node=1 --master_port=29500 tools/train.py configs/metric_learning/imagenet_resnet50_1000kid_jpg.py --work_dir ~/projects/work_dirs/ --launcher pytorch --fp16
```
Comment (wenmengzhou) [inline docs/source/tutorials/metric_learning.md:72]: as tutorial, we can use small dataset like cub200 or car196 or flowers to demo

**Review by wenmengzhou — CHANGES_REQUESTED** [review]

File: configs/classification/resnet/cifar_resnet50.py (hunk 2 above)
Comment (tuofeilunhifi) [reply configs/classification/resnet/cifar_resnet50.py:1]: only reuse model and eval_config

File: configs/metric_learning/cub_resnet50_jpg.py [hunk 7]
```diff
@@ -81,3 +94,6 @@
```
Comment (wenmengzhou) [inline configs/metric_learning/cub_resnet50_jpg.py:94]: metric leanring should use rn50 pretrained models and lr should be smaller such as 0.01 or 0.001, use cos lr policy may be better

File: docs/source/tutorials/EasyCV 度量学习resnet50.ipynb [hunk 8]
```diff
@@ -0,0 +1,333 @@
...
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "62eeff3d-274d-4ff9-8123-8e11d97c0d38",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "! mkdir -p data/cub200 && wget https://data.caltech.edu/tindfiles/serve/1239ea37-e132-42ee-8c09-c383bb54e7ff/ &&  tar -zxf CUB_200_2011.tgz -C data/cub200/"
```
Comment (wenmengzhou) [inline docs/source/tutorials/EasyCV 度量学习resnet50.ipynb:189]: prepare data package in oss://pai-vision-data-hz/EasyCV/datasets/ and provide public-read url

File: docs/source/tutorials/EasyCV度量学习resnet50.ipynb [hunk 9]
```diff
@@ -0,0 +1,333 @@
...
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import cv2\n",
+    "from easycv.predictors.feature_extractor import TorchFeatureExtractor\n",
+    "\n",
+    "output_ckpt = 'work_dirs/metric_learning/cub/r50/best_export.pth'\n",
+    "tcls = TorchFeatureExtractor(output_ckpt)\n",
+    "\n",
+    "img = cv2.imread('aeroplane_s_000004.png')\n",
```
Comment (wenmengzhou) [inline docs/source/tutorials/EasyCV度量学习resnet50.ipynb:304]: should use a test image from cub200

File: configs/metric_learning/cub_resnet50_jpg.py [hunk 10]
```diff
@@ -0,0 +1,107 @@
...
+lr_config = dict(policy='step', step=[30, 60, 90])
+checkpoint_config = dict(interval=10)
+
+# runtime settings
+total_epochs = 90
+
+# export config
+export = dict(export_neck=True)
+
+load_from = '/home/yunji.cjy/pretrain/easycv/resnet50/epoch_100.pth'
```
Comment (wenmengzhou) [inline configs/metric_learning/cub_resnet50_jpg.py:107]: by setting pretrained=True in backbone to enbale user use default pretrained models

**Review by wenmengzhou — APPROVED** [review]
